__tests__/app.test.js

```diff
@@ -154,3 +154,91 @@ test("400: will respond with a 400 when the article_id is invalid", () => {
       expect(body.msg).toBe("Bad Request");
     });
 });
+describe("/api/articles/:article_id/comments", () => {
+  test("POST 201: accepts a request of an object with username and body property, and responds with the posted comment object", () => {
+    const requestBody = {
+      username: "lurker",
+      body: "I am sure, and don't call me Shirley.",
+    };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(requestBody)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.result).toMatchObject({
+          comment_id: expect.any(Number),
+          votes: 0,
+          created_at: expect.any(String),
+          author: "lurker",
+          body: "I am sure, and don't call me Shirley.",
+          article_id: 5,
+        });
+      });
+  });
+});
+test("POST 400: responds with a 400 status code and error message if user inputs an invalid article ID", () => {
+  const requestBody = {
+    username: "lurker",
+    body: "I am sure, and don't call me Shirley.",
+  };
+  return request(app)
+    .post("/api/articles/orange/comments")
+    .send(requestBody)
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad Request");
+    });
+});
+test("POST 400: responds with a 400 status code and error message if user inputs a valid article number but missing post body properties", () => {
+  const requestBody = {};
+  return request(app)
+    .post("/api/articles/5/comments")
+    .send(requestBody)
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Missing Required Fields");
+    });
+});
+describe("/api/articles/:article_id", () => {
+  test("PATCH: 200 should increment a comment vote by input given and respond with the updated comment", () => {
+    const newVote = { inc_votes: 1 };
+    return request(app)
+      .patch("/api/articles/1")
+      .expect(200)
+      .send(newVote)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          article_id: 1,
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+});
+test("PATCH: will respond with a 400 when the article_id is invalid", () => {
+  const newVote = { inc_votes: 1 };
+
+  return request(app)
+    .patch("/api/articles/99999")
+    .send(newVote)
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Article Not Found");
+    });
+});
+test("PATCH: responds with a 400 status code and error message if user inputs an invalid article ID ", () => {
+  const newVote = { inc_votes: 1 };
+
+  return request(app)
+    .patch("/api/articles/banana")
+    .send(newVote)
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad Request");
+    });
```

Comment:
Really good error handling for the article id, but dont forget to add a error handling test when inc_votes is missing/invalid